src/main/js/screens/app/screens/select/selectScreen.js

```diff
@@ -136,89 +135,167 @@ export class selectScreen extends React.Component {
   }
 
   handleStarredClick(lesson) {
-    return this.props.starredLessons.map(userLesson => userLesson.lesson.name).includes(lesson.name)
-      ? this.props.removeStarredLesson(lesson.name)
-      : this.props.addStarredLesson(lesson.name);
+    this.props.starredLessons.map(userLesson => userLesson.lesson.name).includes(lesson.name)
+      ? this.props.removeStarredLesson(lesson.name, this.props.params.type)
+      : this.props.addStarredLesson(lesson.name, this.props.params.type);
   }
 
   getNumberOfQuestions(lesson) {
-    if (this.props.spacedRepetitionModes.includes(this.props.params.type) || this.props.params.type === 'grammar') {
+    if (
+      this.props.addressedQuestionsInLessons &&
+      this.props.addressedQuestionsInLessons[lesson.name] &&
+      this.props.spacedRepetitionModes.includes(this.props.params.type)
+    ) {
       const { unanswered, retention, all } = this.props.addressedQuestionsInLessons[lesson.name];
       return { unanswered, retention, all };
     }
-    return { unanswered: 0, retention: 0, total: 0 };
+    return { unanswered: 0, retention: 0, all: 0 };
+  }
+
+  getNumberOfFavoriteQuestions() {
+    if (
+      this.props.favoriteLesson &&
+      this.props.favoriteLesson.nuggetData &&
+      this.props.spacedRepetitionModes.includes(this.props.params.type)
+    ) {
+      const { unanswered, retention, all } = this.props.favoriteLesson.nuggetData;
+      return { unanswered, retention, all };
+    }
+    return { unanswered: 0, retention: 0, all: 0 };
```

Comment:
See if getNumberOfFavoriteQuestions and getNumberOfQuestions can use a shared function to retrieve data.